Цель работы: необходимо реализовать базовый анализ датасета: вывести весь датасет и описать его, вывести размерноть массива, вывести наименование всех колонок, написать что они означают в датасете, вывести все уникальные значения,отсортировать по определенным параметрам, удалить ненужные столбцы или строчки, заменить определенные значения в датасете , удалить дубликаты, провести анализ с помощью функций info, провести анализ с помощью функций describe, провести выборку данных по строкам и столбцам с помощью loc, сохранить новый датасет, сделать вывод по каждому пункту и описать все.

In [1]:
import pandas as pd
import numpy as np
waterQuality1 = pd.read_csv("waterQuality1.csv") 
waterQuality1

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

1. Выводим весь датасет. В данном датасете (количество компонентов в воде) находятся следующие данные: алюминий, аммиак, мышьяк, барий, кадмий, хлорамин, бактерии, фторид, хром, медь, нитраты, свинец, нитриты , перхлорат, ртуть, радий, селен, серебро, уран, безопасность (количсетво данных веществ в воде).

In [28]:
waterQuality1.shape

(7999, 21)

2. Выводим размерность массива

In [5]:
waterQuality1.columns

Index(['aluminium', 'ammonia', 'arsenic', 'barium', 'cadmium', 'chloramine',
       'chromium', 'copper', 'flouride', 'bacteria', 'viruses', 'lead',
       'nitrates', 'nitrites', 'mercury', 'perchlorate', 'radium', 'selenium',
       'silver', 'uranium', 'is_safe'],
      dtype='object')

3. Выводим наименование всех колонок и то, что они значат: алюминий, аммиак, мышьяк, барий, кадмий, хлорамин, бактерии, фторид, хром, медь, нитраты, свинец, нитриты , перхлорат, ртуть, радий, селен, серебро, уран, безопасность. 

In [6]:
waterQuality1["aluminium"].unique()

array([1.65, 2.32, 1.01, 1.36, 0.92, 0.94, 2.36, 3.93, 0.6 , 0.22, 3.27,
       1.35, 1.88, 4.93, 2.89, 0.61, 3.47, 2.11, 4.88, 4.12, 0.68, 1.15,
       0.27, 4.32, 3.31, 1.82, 3.42, 4.41, 4.57, 1.69, 1.87, 2.71, 1.63,
       0.01, 4.49, 3.52, 4.35, 2.51, 2.34, 1.71, 3.45, 3.37, 3.13, 0.29,
       1.51, 1.09, 2.68, 0.3 , 0.23, 2.55, 0.86, 0.24, 3.63, 3.26, 0.42,
       0.77, 0.58, 3.82, 2.78, 3.62, 3.88, 4.64, 2.25, 0.54, 4.62, 2.5 ,
       3.08, 3.03, 2.85, 1.38, 2.73, 3.02, 1.96, 2.35, 2.45, 2.67, 1.12,
       3.3 , 0.43, 1.19, 0.85, 0.5 , 4.75, 1.89, 3.75, 3.32, 1.75, 2.66,
       2.76, 2.63, 0.04, 0.88, 2.22, 4.91, 4.8 , 0.47, 1.62, 0.09, 4.66,
       2.74, 1.79, 4.15, 3.78, 2.91, 4.3 , 3.24, 3.53, 3.04, 2.21, 2.47,
       1.92, 1.22, 2.12, 4.19, 4.21, 4.07, 2.56, 3.55, 3.94, 0.8 , 2.59,
       3.9 , 3.15, 0.18, 3.86, 1.25, 2.29, 0.07, 4.18, 4.34, 2.39, 0.9 ,
       0.89, 2.4 , 3.69, 4.71, 3.36, 4.46, 4.5 , 0.57, 2.69, 2.17, 1.32,
       5.  , 2.6 , 0.35, 3.83, 2.9 , 1.56, 2.23, 3.

In [8]:
waterQuality1["ammonia"].unique()

array(['9.08', '21.16', '14.02', ..., '2.78', '27.12', '10'], dtype=object)

In [9]:
waterQuality1["arsenic"].unique()

array([4.00e-02, 1.00e-02, 3.00e-02, 2.00e-02, 1.00e-03, 5.00e-02,
       3.10e-01, 3.50e-01, 4.80e-01, 8.70e-01, 1.70e-01, 2.10e-01,
       8.10e-01, 8.80e-01, 6.60e-01, 7.00e-01, 5.40e-01, 4.00e-01,
       5.70e-01, 3.70e-01, 1.00e-01, 9.90e-01, 2.50e-01, 7.70e-01,
       2.90e-01, 4.90e-01, 5.90e-01, 6.50e-01, 6.80e-01, 3.00e-01,
       9.30e-01, 8.40e-01, 1.60e-01, 4.60e-01, 1.50e-01, 2.20e-01,
       7.10e-01, 7.90e-01, 8.00e-02, 5.30e-01, 3.60e-01, 5.20e-01,
       6.00e-02, 9.20e-01, 5.60e-01, 4.40e-01, 3.90e-01, 6.70e-01,
       8.60e-01, 7.50e-01, 9.60e-01, 8.00e-01, 8.90e-01, 7.40e-01,
       4.70e-01, 4.30e-01, 5.10e-01, 1.20e-01, 7.80e-01, 6.30e-01,
       7.60e-01, 3.30e-01, 1.80e-01, 9.10e-01, 9.50e-01, 6.10e-01,
       7.30e-01, 2.70e-01, 1.03e+00, 1.01e+00, 9.00e-02, 2.80e-01,
       6.00e-01, 1.04e+00, 9.80e-01, 5.80e-01, 7.00e-02, 6.20e-01,
       8.50e-01, 3.80e-01, 4.10e-01, 9.70e-01, 2.60e-01, 3.40e-01,
       1.00e+00, 2.00e-01, 1.10e-01, 8.30e-01, 4.20e-01, 1.40e

In [10]:
waterQuality1["barium"].unique()

array([2.85, 3.31, 0.58, 2.96, 0.2 , 2.88, 1.35, 0.66, 0.71, 1.37, 2.69,
       0.84, 2.78, 3.05, 3.77, 0.59, 0.06, 0.88, 0.36, 3.43, 0.04, 0.97,
       0.55, 2.6 , 0.68, 0.46, 0.85, 2.8 , 1.76, 3.04, 2.01, 3.74, 3.02,
       2.54, 2.93, 1.87, 1.58, 2.67, 3.63, 1.86, 0.25, 0.52, 2.16, 3.94,
       3.98, 2.5 , 3.5 , 1.31, 1.98, 1.07, 2.79, 3.86, 0.65, 3.45, 3.53,
       1.71, 0.22, 2.27, 3.11, 0.57, 2.24, 3.92, 1.67, 1.47, 1.78, 3.76,
       2.86, 3.25, 0.07, 3.08, 1.28, 0.75, 0.17, 2.97, 2.99, 3.18, 1.38,
       1.36, 3.54, 2.09, 2.  , 3.28, 1.74, 2.44, 0.92, 0.13, 2.73, 3.2 ,
       0.29, 1.46, 1.32, 3.61, 0.16, 3.84, 2.36, 1.5 , 2.46, 1.99, 0.76,
       2.66, 2.9 , 1.6 , 2.34, 1.4 , 1.01, 1.11, 3.65, 1.65, 3.29, 1.94,
       3.16, 1.8 , 2.63, 0.86, 3.64, 0.87, 3.15, 1.03, 1.82, 1.39, 1.14,
       3.58, 3.35, 3.57, 1.81, 1.63, 0.56, 3.93, 1.72, 0.28, 3.09, 1.93,
       0.12, 2.42, 3.33, 2.53, 2.57, 0.78, 1.84, 3.03, 3.51, 0.19, 3.34,
       2.91, 3.48, 0.33, 2.39, 2.07, 2.22, 1.18, 0.

In [11]:
waterQuality1["cadmium"].unique()

array([0.007, 0.002, 0.008, 0.001, 0.006, 0.003, 0.004, 0.005, 0.009,
       0.01 , 0.08 , 0.04 , 0.03 , 0.07 , 0.09 , 0.12 , 0.1  , 0.02 ,
       0.05 , 0.11 , 0.   , 0.06 , 0.13 ])

In [12]:
waterQuality1["chloramine"].unique()

array([0.35, 5.28, 4.24, 7.23, 2.67, 0.8 , 1.28, 6.22, 3.14, 6.4 , 5.75,
       0.1 , 0.05, 0.7 , 5.99, 1.94, 5.29, 7.78, 1.21, 0.01, 4.57, 3.47,
       3.74, 7.24, 4.07, 7.22, 2.55, 2.87, 6.63, 2.78, 5.87, 2.86, 4.  ,
       4.25, 7.75, 4.55, 4.94, 0.64, 4.3 , 2.24, 2.4 , 2.9 , 3.58, 2.99,
       5.71, 1.26, 5.38, 0.22, 6.11, 0.68, 7.7 , 5.47, 7.97, 6.19, 4.08,
       2.49, 0.9 , 4.5 , 0.88, 1.13, 3.85, 6.16, 7.06, 3.43, 4.4 , 4.89,
       6.57, 2.15, 2.38, 4.77, 7.81, 2.79, 1.78, 1.97, 3.4 , 4.41, 5.27,
       3.35, 2.69, 6.06, 7.56, 3.55, 0.13, 7.52, 3.57, 6.02, 1.12, 6.1 ,
       0.19, 1.85, 3.32, 1.89, 0.46, 4.8 , 4.59, 6.25, 4.54, 5.22, 1.86,
       7.37, 7.36, 3.07, 2.89, 6.75, 3.2 , 3.11, 1.33, 4.43, 6.6 , 0.63,
       5.07, 5.14, 0.4 , 4.13, 2.52, 3.71, 1.87, 7.09, 2.26, 0.41, 3.42,
       5.65, 5.72, 0.89, 1.57, 2.74, 0.79, 7.8 , 3.49, 3.83, 3.81, 1.69,
       6.21, 1.07, 0.5 , 3.95, 7.59, 0.98, 4.95, 2.45, 1.36, 6.5 , 4.68,
       5.1 , 3.61, 4.23, 0.81, 4.06, 4.42, 7.95, 6.

In [13]:
waterQuality1["chromium"].unique()

array([0.83, 0.68, 0.53, 0.03, 0.69, 0.43, 0.62, 0.1 , 0.77, 0.49, 0.15,
       0.76, 0.42, 0.51, 0.54, 0.47, 0.88, 0.41, 0.2 , 0.65, 0.12, 0.61,
       0.13, 0.73, 0.25, 0.72, 0.78, 0.52, 0.17, 0.74, 0.32, 0.87, 0.26,
       0.9 , 0.36, 0.84, 0.11, 0.85, 0.09, 0.35, 0.82, 0.04, 0.81, 0.64,
       0.23, 0.45, 0.02, 0.89, 0.71, 0.6 , 0.29, 0.01, 0.31, 0.22, 0.18,
       0.33, 0.38, 0.79, 0.19, 0.46, 0.7 , 0.3 , 0.24, 0.16, 0.63, 0.39,
       0.34, 0.08, 0.8 , 0.4 , 0.14, 0.27, 0.  , 0.59, 0.28, 0.07, 0.37,
       0.21, 0.86, 0.05, 0.66, 0.55, 0.67, 0.06, 0.56, 0.58, 0.57, 0.48,
       0.5 , 0.75, 0.44])

In [14]:
waterQuality1["copper"].unique()

array([0.17, 0.66, 0.02, 1.66, 0.57, 1.38, 1.88, 1.86, 1.45, 0.82, 0.6 ,
       1.  , 1.35, 0.79, 1.54, 1.08, 1.15, 0.71, 1.82, 1.18, 1.51, 1.77,
       1.23, 1.34, 1.05, 1.09, 0.27, 1.57, 0.46, 1.53, 1.03, 1.56, 0.9 ,
       1.85, 0.91, 0.08, 0.22, 0.29, 1.92, 1.29, 1.14, 1.71, 1.59, 0.3 ,
       0.54, 1.79, 0.44, 1.97, 1.98, 0.65, 1.06, 0.14, 1.26, 1.69, 1.75,
       1.61, 0.19, 0.38, 0.34, 0.93, 0.7 , 1.55, 1.11, 0.42, 0.18, 1.28,
       1.99, 1.21, 1.3 , 0.95, 0.01, 1.44, 0.52, 1.01, 1.64, 0.76, 1.5 ,
       0.25, 0.87, 0.73, 1.1 , 0.77, 0.62, 0.36, 1.43, 1.65, 0.05, 0.16,
       0.13, 1.72, 1.67, 1.89, 0.94, 0.85, 1.7 , 0.72, 1.52, 0.24, 1.16,
       1.94, 1.95, 1.4 , 0.55, 1.39, 0.37, 0.35, 0.5 , 1.24, 1.04, 0.33,
       0.81, 1.22, 0.78, 0.04, 1.8 , 1.83, 1.37, 0.75, 0.07, 1.96, 1.07,
       0.06, 0.21, 1.33, 0.89, 0.88, 1.32, 0.98, 0.53, 1.36, 0.03, 1.74,
       1.41, 1.6 , 1.17, 0.59, 0.11, 1.46, 1.84, 0.68, 0.32, 0.8 , 1.76,
       0.67, 1.68, 0.28, 0.56, 1.49, 1.25, 0.39, 1.

In [15]:
waterQuality1["flouride"].unique()

array([0.05, 0.9 , 0.99, 1.08, 0.61, 0.11, 0.33, 0.86, 0.98, 1.24, 1.29,
       0.58, 0.09, 1.07, 0.54, 0.62, 1.43, 0.34, 0.22, 1.  , 1.46, 0.43,
       1.44, 0.29, 1.35, 0.53, 0.26, 1.41, 0.87, 0.51, 0.91, 0.  , 0.55,
       0.42, 0.4 , 0.68, 1.2 , 1.04, 0.36, 1.42, 0.52, 0.14, 0.89, 1.33,
       0.59, 0.66, 0.75, 0.1 , 0.15, 1.01, 0.83, 0.03, 1.47, 1.28, 0.45,
       0.02, 0.2 , 0.32, 1.09, 1.13, 0.13, 1.14, 1.21, 0.5 , 0.28, 0.3 ,
       0.01, 1.16, 0.69, 1.25, 1.48, 0.76, 0.97, 0.73, 1.49, 0.08, 1.18,
       0.25, 1.06, 0.16, 0.04, 1.27, 1.38, 1.12, 0.74, 0.93, 0.92, 1.39,
       0.07, 0.24, 0.48, 0.37, 0.85, 1.4 , 1.03, 0.65, 0.44, 0.7 , 0.79,
       0.82, 1.5 , 1.19, 0.78, 1.34, 0.96, 0.8 , 0.19, 1.45, 1.23, 1.37,
       0.49, 0.23, 1.26, 0.95, 0.88, 1.32, 0.17, 0.84, 1.3 , 0.64, 0.21,
       0.81, 0.12, 0.94, 1.05, 1.11, 0.63, 0.57, 1.31, 0.67, 0.77, 0.46,
       0.06, 0.38, 1.22, 0.72, 1.36, 0.31, 1.02, 0.27, 0.47, 1.1 , 0.41,
       1.15, 0.71, 0.18, 0.56, 0.6 , 0.35, 1.17, 0.

In [16]:
waterQuality1["bacteria"].unique()

array([0.2 , 0.65, 0.05, 0.71, 0.13, 0.67, 0.16, 0.35, 0.83, 0.04, 0.52,
       0.91, 0.7 , 0.23, 0.89, 0.85, 0.75, 0.99, 0.92, 0.58, 0.8 , 0.56,
       0.96, 0.25, 0.44, 0.69, 0.08, 0.77, 0.38, 0.12, 0.3 , 0.26, 0.63,
       0.86, 0.32, 0.49, 0.27, 0.18, 0.06, 0.79, 0.84, 0.42, 0.98, 0.24,
       0.74, 0.76, 0.51, 0.72, 0.07, 0.53, 0.45, 0.39, 0.1 , 0.31, 0.11,
       0.64, 0.57, 0.82, 0.87, 0.55, 0.14, 1.  , 0.47, 0.33, 0.09, 0.28,
       0.19, 0.78, 0.02, 0.41, 0.46, 0.61, 0.62, 0.5 , 0.59, 0.15, 0.48,
       0.68, 0.01, 0.73, 0.22, 0.21, 0.34, 0.43, 0.93, 0.66, 0.36, 0.17,
       0.6 , 0.37, 0.29, 0.97, 0.95, 0.9 , 0.03, 0.4 , 0.94, 0.54, 0.81,
       0.88, 0.  ])

In [17]:
waterQuality1["viruses"].unique()

array([0.   , 0.65 , 0.003, 0.71 , 0.001, 0.67 , 0.007, 0.005, 0.002,
       0.83 , 0.008, 0.52 , 0.91 , 0.7  , 0.009, 0.89 , 0.85 , 0.75 ,
       0.99 , 0.92 , 0.58 , 0.8  , 0.56 , 0.96 , 0.69 , 0.006, 0.77 ,
       0.004, 0.63 , 0.86 , 0.79 , 0.84 , 0.98 , 0.74 , 0.76 , 0.51 ,
       0.72 , 0.53 , 0.64 , 0.57 , 0.82 , 0.87 , 0.55 , 1.   , 0.78 ,
       0.61 , 0.62 , 0.5  , 0.59 , 0.68 , 0.73 , 0.93 , 0.66 , 0.6  ,
       0.97 , 0.95 , 0.9  , 0.94 , 0.54 , 0.81 , 0.88 ])

In [18]:
waterQuality1["lead"].unique()

array([0.054, 0.1  , 0.078, 0.016, 0.117, 0.135, 0.021, 0.197, 0.167,
       0.109, 0.145, 0.011, 0.103, 0.101, 0.126, 0.017, 0.08 , 0.065,
       0.071, 0.108, 0.086, 0.061, 0.114, 0.012, 0.031, 0.11 , 0.182,
       0.049, 0.079, 0.013, 0.193, 0.091, 0.124, 0.107, 0.085, 0.146,
       0.162, 0.048, 0.082, 0.141, 0.131, 0.069, 0.136, 0.166, 0.112,
       0.169, 0.093, 0.025, 0.139, 0.184, 0.015, 0.09 , 0.068, 0.123,
       0.177, 0.149, 0.062, 0.121, 0.105, 0.168, 0.137, 0.189, 0.095,
       0.053, 0.113, 0.132, 0.175, 0.045, 0.036, 0.033, 0.191, 0.127,
       0.032, 0.052, 0.047, 0.019, 0.129, 0.153, 0.147, 0.12 , 0.003,
       0.16 , 0.18 , 0.072, 0.042, 0.092, 0.023, 0.185, 0.07 , 0.083,
       0.155, 0.164, 0.17 , 0.043, 0.198, 0.194, 0.089, 0.044, 0.087,
       0.122, 0.088, 0.158, 0.074, 0.148, 0.172, 0.041, 0.176, 0.038,
       0.005, 0.181, 0.159, 0.05 , 0.156, 0.034, 0.138, 0.102, 0.064,
       0.157, 0.008, 0.059, 0.037, 0.118, 0.116, 0.06 , 0.075, 0.04 ,
       0.024, 0.144,

In [19]:
waterQuality1["nitrates"].unique()

array([16.08,  2.01, 14.16, ..., 12.76, 19.8 ,  0.  ])

In [20]:
waterQuality1["nitrites"].unique()

array([1.13, 1.93, 1.11, 1.29, 1.89, 1.78, 1.81, 1.84, 1.46, 1.25, 1.49,
       1.95, 1.82, 1.08, 1.2 , 1.53, 1.22, 1.76, 1.41, 1.5 , 1.18, 1.74,
       1.92, 1.07, 1.7 , 1.69, 1.43, 1.19, 1.55, 1.  , 1.27, 1.86, 1.35,
       1.56, 1.83, 1.68, 1.73, 1.09, 1.28, 1.34, 1.21, 1.87, 1.45, 1.24,
       1.79, 1.37, 1.48, 1.16, 1.4 , 1.57, 1.88, 1.59, 1.36, 1.9 , 1.06,
       1.04, 1.58, 1.31, 1.38, 1.17, 1.3 , 1.02, 1.65, 1.47, 1.14, 1.51,
       1.66, 1.99, 1.26, 1.91, 1.42, 1.33, 1.97, 1.03, 1.8 , 1.1 , 1.05,
       1.64, 1.44, 1.15, 1.63, 1.72, 1.23, 1.85, 1.67, 1.6 , 1.54, 1.94,
       1.12, 1.62, 1.52, 1.32, 1.98, 1.77, 1.61, 1.01, 1.75, 1.71, 2.  ,
       1.96, 1.39, 2.12, 2.43, 2.03, 2.08, 2.06, 2.09, 2.19, 2.01, 2.27,
       2.38, 2.02, 2.36, 2.44, 2.22, 2.21, 2.72, 2.25, 2.24, 2.28, 2.07,
       2.2 , 2.62, 2.18, 2.04, 2.49, 2.29, 2.68, 2.56, 2.23, 2.58, 2.17,
       2.05, 2.16, 2.53, 2.3 , 2.35, 2.11, 2.75, 2.26, 2.89, 2.52, 2.86,
       2.34, 2.1 , 2.63, 2.37, 2.33, 2.14, 2.42, 2.

In [21]:
waterQuality1["mercury"].unique()

array([0.007, 0.003, 0.006, 0.004, 0.001, 0.01 , 0.009, 0.008, 0.002,
       0.005, 0.   ])

In [22]:
waterQuality1["perchlorate"].unique()

array([37.75, 32.26, 50.28, ..., 12.74,  9.42,  8.44])

In [23]:
waterQuality1["radium"].unique()

array([6.78, 3.21, 7.07, 1.72, 2.41, 5.42, 2.84, 7.24, 4.99, 0.08, 7.8 ,
       1.3 , 1.97, 5.58, 4.33, 0.98, 6.89, 1.14, 1.  , 0.88, 3.05, 1.74,
       0.9 , 3.22, 7.99, 0.49, 1.15, 3.29, 2.39, 4.55, 7.77, 2.21, 6.16,
       1.71, 3.76, 6.15, 3.19, 0.5 , 4.7 , 6.14, 5.45, 0.37, 3.39, 2.18,
       2.3 , 2.67, 4.27, 2.91, 4.12, 0.92, 1.53, 7.12, 5.63, 5.8 , 7.18,
       4.87, 4.86, 4.34, 2.75, 3.81, 6.13, 4.91, 4.39, 6.65, 3.75, 0.63,
       1.44, 7.4 , 7.81, 2.95, 1.29, 2.73, 6.35, 6.53, 7.02, 2.92, 2.35,
       6.11, 0.86, 0.28, 5.14, 5.41, 4.15, 2.24, 6.46, 4.74, 1.08, 4.29,
       6.98, 3.95, 3.32, 1.62, 7.6 , 6.6 , 1.2 , 6.36, 4.96, 5.77, 4.48,
       6.74, 4.41, 6.28, 7.28, 1.45, 5.26, 2.2 , 0.25, 3.4 , 4.23, 4.14,
       3.09, 2.58, 3.49, 1.39, 3.04, 0.68, 7.9 , 5.99, 4.8 , 5.11, 2.4 ,
       2.53, 2.9 , 6.59, 3.42, 4.16, 1.23, 5.9 , 7.25, 6.42, 3.44, 1.67,
       2.65, 6.2 , 6.66, 4.35, 3.24, 7.35, 3.23, 4.63, 6.49, 1.31, 3.72,
       6.75, 6.63, 4.47, 7.42, 7.85, 4.05, 0.77, 2.

In [24]:
waterQuality1["selenium"].unique()

array([0.08, 0.07, 0.02, 0.1 , 0.03, 0.05, 0.09, 0.01, 0.06, 0.  , 0.04])

In [25]:
waterQuality1["silver"].unique()

array([0.34, 0.27, 0.44, 0.45, 0.06, 0.19, 0.24, 0.08, 0.25, 0.31, 0.33,
       0.48, 0.38, 0.05, 0.47, 0.12, 0.4 , 0.41, 0.1 , 0.13, 0.01, 0.16,
       0.18, 0.5 , 0.26, 0.17, 0.2 , 0.11, 0.3 , 0.37, 0.35, 0.04, 0.46,
       0.  , 0.43, 0.42, 0.03, 0.28, 0.02, 0.21, 0.32, 0.09, 0.15, 0.14,
       0.29, 0.22, 0.39, 0.36, 0.23, 0.07, 0.49])

In [26]:
waterQuality1["uranium"].unique()

array([0.02, 0.05, 0.01, 0.08, 0.07, 0.06, 0.03, 0.  , 0.04, 0.09])

In [27]:
waterQuality1["is_safe"].unique()

array(['1', '0', '#NUM!'], dtype=object)

4. Выводим все уникальные значения в каждой из колонок в датасете.

In [29]:
waterQuality1.sort_values(by='uranium')

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
2118       0.13   16.35    0.030    2.90    0.008        5.14      0.29   
5740       0.06    5.22    0.040    0.20    0.030        0.08      0.03   
5736       0.10   24.24    0.080    0.40    0.020        0.17      0.01   
2878       0.03   24.89    0.001    2.40    0.005        4.90      0.59   
1842       0.55    8.06    0.960    2.72    0.040        0.66      0.88   
...         ...     ...      ...     ...      ...         ...       ...   
4493       0.06   23.89    0.050    0.32    0.060        0.33      0.09   
4492       0.03   13.15    0.020    1.42    0.080        0.01      0.02   
1272       3.78    4.59    0.370    0.47    0.040        1.99      0.69   
7350       0.09   26.46    0.020    0.91    0.090        0.53      0.05   
2120       0.29    5.43    0.040    2.34    0.002        4.13      0.70   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
2118    1.50      0.25      0.02  ...  0.083      6.22      1.91    0.009   
5740    0.03      0.15      0.09  ...  0.165     17.06      0.56    0.003   
5736    1.65      0.69      0.74  ...  0.046     12.31      0.53    0.003   
2878    1.07      1.04      0.76  ...  0.043      2.65      1.87    0.002   
1842    0.69      1.32      0.62  ...  0.103     13.87      1.57    0.010   
...      ...       ...       ...  ...    ...       ...       ...      ...   
4493    0.08      0.21      0.00  ...  0.092      5.21      0.78    0.010   
4492    0.01      0.10      0.00  ...  0.165     10.82      0.53    0.006   
1272    1.49      0.75      0.00  ...  0.099     10.50      1.85    0.007   
7350    1.02      1.24      0.44  ...  0.197     11.14      1.55    0.002   
2120    1.51      0.89      0.55  ...  0.164      8.49      1.49    0.001   

      perchlorate  radium  selenium  silver  uranium  is_safe  
2118        50.04    1.39      0.02    0.11     0.00        0  
5740         0.15    0.18      0.01    0.01     0.00        0  
5736         0.48    0.92      0.01    0.06     0.00        0  
2878        35.64    3.13      0.08    0.27     0.00        0  
1842        15.66    5.29      0.06    0.46     0.00        0  
...           ...     ...       ...     ...      ...      ...  
4493         5.86    3.07      0.08    0.00     0.09        0  
4492         0.17    0.72      0.07    0.08     0.09        0  
1272        11.27    6.53      0.02    0.07     0.09        0  
7350         1.70    2.62      0.08    0.02     0.09        0  
2120        37.15    0.25      0.08    0.01     0.09        0  

[7999 rows x 21 columns]

5. Отсортировка по параметру уран - "uranium", в котором произошла отсортировака по возрастаню от минимального значения - 0.0 до максимального значения - 0.09 .


In [32]:

waterQuality1.sort_values(by='silver', ascending=False)

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
2903       0.09    8.11     0.01    1.47    0.007        3.27      0.77   
693        4.59    3.44     0.01    2.55    0.009        5.17      0.37   
2122       0.15   29.72     0.02    1.40    0.008        3.71      0.47   
3122       0.02   29.62     0.34    2.10    0.040        3.81      0.47   
3564       0.08    0.41     0.57    2.66    0.080        1.17      0.33   
...         ...     ...      ...     ...      ...         ...       ...   
6731       0.02   10.42     0.04    2.48    0.090        0.20      0.00   
4987       0.00      26     0.08    0.31    0.040        0.02      0.05   
6737       0.01   18.88     0.02    0.99    0.060        0.00      0.04   
2370       0.06   19.59     0.04    0.79    0.007        6.30      0.31   
3999       0.05    8.83     0.00    0.97    0.080        0.09      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
2903    1.56      1.47      0.97  ...  0.166     12.39      1.29    0.005   
693     1.20      1.45      0.00  ...  0.091     13.66      1.66    0.001   
2122    1.16      1.06      0.82  ...  0.120      6.63      1.57    0.001   
3122    1.16      1.06      0.82  ...  0.120      6.98      2.22    0.001   
3564    0.06      1.49      0.15  ...  0.001     10.11      1.85    0.007   
...      ...       ...       ...  ...    ...       ...       ...      ...   
6731    0.07      0.73      0.06  ...  0.157     10.83      1.84    0.006   
4987    1.62      0.96      0.00  ...  0.138      5.66      0.34    0.000   
6737    1.26      1.04      0.17  ...  0.200     16.09      1.23    0.005   
2370    1.91      0.96      0.14  ...  0.132     16.05      1.09    0.010   
3999    0.03      0.05      0.00  ...  0.054     16.08      0.93    0.007   

      perchlorate  radium  selenium  silver  uranium  is_safe  
2903        47.39    3.54      0.05     0.5     0.01        0  
693         31.78    2.07      0.04     0.5     0.09        1  
2122         4.28    7.90      0.04     0.5     0.04        0  
3122        46.16    7.90      0.04     0.5     0.04        0  
3564        53.45    2.54      0.04     0.5     0.09        0  
...           ...     ...       ...     ...      ...      ...  
6731         3.33    0.10      0.03     0.0     0.04        0  
4987         0.44    0.52      0.01     0.0     0.06        0  
6737         0.09    2.62      0.06     0.0     0.05        0  
2370        49.72    5.19      0.02     0.0     0.02        0  
3999         3.57    0.28      0.08     0.0     0.02        0  

[7999 rows x 21 columns]

5. Отсортировка по параметру серебро - "silver", в котором произошла отсортировака по убыванию от максимального значения - 0.5 до минимального значения - 0.0 

In [34]:
waterQuality1.drop(columns = ["nitrites"])

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  viruses   lead  nitrates  mercury  \
0       0.17      0.05      0.20    0.000  0.054     16.08    0.007   
1       0.66      0.90      0.65    0.650  0.100      2.01    0.003   
2       0.02      0.99      0.05    0.003  0.078     14.16    0.006   
3       1.66      1.08      0.71    0.710  0.016      1.41    0.004   
4       0.57      0.61      0.13    0.001  0.117      6.74    0.003   
...      ...       ...       ...      ...    ...       ...      ...   
7994    0.03      1.37      0.00    0.000  0.197     14.29    0.005   
7995    0.02      1.48      0.00    0.000  0.031     10.27    0.001   
7996    0.02      0.91      0.00    0.000  0.182     15.92    0.000   
7997    0.09      0.00      0.00    0.000  0.000      0.00    0.000   
7998    0.03      1.00      0.00    0.000  0.182     15.92    0.000   

      perchlorate  radium  selenium  silver  uranium is_safe  
0           37.75    6.78      0.08    0.34     0.02       1  
1           32.26    3.21      0.08    0.27     0.05       1  
2           50.28    7.07      0.07    0.44     0.01       0  
3            9.12    1.72      0.02    0.45     0.05       1  
4           16.90    2.41      0.02    0.06     0.02       1  
...           ...     ...       ...     ...      ...     ...  
7994         3.57    2.13      0.09    0.06     0.03       1  
7995         1.48    1.11      0.09    0.10     0.08       1  
7996         1.35    4.84      0.00    0.04     0.05       1  
7997         0.00    0.00      0.00    0.00     0.00       1  
7998         1.35    4.84      0.00    0.04     0.05       1  

[7999 rows x 20 columns]

6. Удаление колонки с именем "nitrites".

In [38]:
waterQuality1.drop([1,2], axis=0, inplace=True)
waterQuality1

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
5          0.94   14.47     0.03    2.88    0.003        0.80      0.43   
6          2.36     5.6     0.01    1.35    0.004        1.28      0.62   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
5       1.38      0.11      0.67  ...  0.135      9.75      1.89    0.006   
6       1.88      0.33      0.13  ...  0.021     18.60      1.78    0.007   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
5           27.17    5.42      0.08    0.19     0.02        1  
6           45.34    2.84      0.10    0.24     0.08        0  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7997 rows x 21 columns]

6. Удаление строк с индексами 1 и 2 во всм датасете.

In [62]:
waterQuality1=waterQuality1.replace({"perchlorate":9.12},10)
waterQuality1

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
5          0.94   14.47     0.03    2.88    0.003        0.80      0.43   
6          2.36     5.6     0.01    1.35    0.004        1.28      0.62   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
5       1.38      0.11      0.67  ...  0.135      9.75      1.89    0.006   
6       1.88      0.33      0.13  ...  0.021     18.60      1.78    0.007   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
3           10.00    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
5           27.17    5.42      0.08    0.19     0.02        1  
6           45.34    2.84      0.10    0.24     0.08        0  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7997 rows x 21 columns]

7.Заменена всех значений 9.12 в колонке perchlorate на значение 10.

In [53]:
waterQuality1.drop_duplicates()

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
5          0.94   14.47     0.03    2.88    0.003        0.80      0.43   
6          2.36     5.6     0.01    1.35    0.004        1.28      0.62   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
5       1.38      0.11      0.67  ...  0.135      9.75      1.89    0.006   
6       1.88      0.33      0.13  ...  0.021     18.60      1.78    0.007   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
5           27.17    5.42      0.08    0.19     0.02        1  
6           45.34    2.84      0.10    0.24     0.08        0  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7997 rows x 21 columns]

8. Удаление дубликатов во всем датасете, которое позволяет нам читабельность таблицы сделать лучше.

In [54]:
waterQuality1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 7997 entries, 0 to 7998
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   aluminium    7997 non-null   float64
 1   ammonia      7997 non-null   object 
 2   arsenic      7997 non-null   float64
 3   barium       7997 non-null   float64
 4   cadmium      7997 non-null   float64
 5   chloramine   7997 non-null   float64
 6   chromium     7997 non-null   float64
 7   copper       7997 non-null   float64
 8   flouride     7997 non-null   float64
 9   bacteria     7997 non-null   float64
 10  viruses      7997 non-null   float64
 11  lead         7997 non-null   float64
 12  nitrates     7997 non-null   float64
 13  nitrites     7997 non-null   float64
 14  mercury      7997 non-null   float64
 15  perchlorate  7997 non-null   float64
 16  radium       7997 non-null   float64
 17  selenium     7997 non-null   float64
 18  silver       7997 non-null   float64
 19  uranium    

9. Проведен анализ с помощью функций info, которая позволяет вывести общую информацию о дататайме, к примеру, функция вывела, что здесь находится 21 столбец, типы внесенных данных в таблицу и каково их кол-во, 7997 строк.

In [55]:
waterQuality1.describe()

aluminium      arsenic       barium      cadmium   chloramine  \
count  7997.000000  7997.000000  7997.000000  7997.000000  7997.000000   
mean      0.665908     0.161479     1.567620     0.042816     2.176185   
std       1.265162     0.252613     1.216037     0.036049     2.567010   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.040000     0.030000     0.560000     0.008000     0.100000   
50%       0.070000     0.050000     1.190000     0.040000     0.530000   
75%       0.280000     0.100000     2.480000     0.070000     4.240000   
max       5.050000     1.050000     4.940000     0.130000     8.680000   

          chromium       copper     flouride     bacteria      viruses  \
count  7997.000000  7997.000000  7997.000000  7997.000000  7997.000000   
mean      0.247136     0.805973     0.771522     0.319657     0.328583   
std       0.270612     0.653560     0.435418     0.329492     0.378109   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.050000     0.090000     0.400000     0.000000     0.002000   
50%       0.090000     0.750000     0.770000     0.220000     0.008000   
75%       0.440000     1.390000     1.160000     0.610000     0.700000   
max       0.900000     2.000000     1.500000     1.000000     1.000000   

              lead     nitrates     nitrites      mercury  perchlorate  \
count  7997.000000  7997.000000  7997.000000  7997.000000  7997.000000   
mean      0.099452     9.819256     1.329914     0.005194    16.454094   
std       0.058179     5.541124     0.573247     0.002967    17.684759   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.048000     5.000000     1.000000     0.003000     2.170000   
50%       0.102000     9.930000     1.420000     0.005000     7.740000   
75%       0.151000    14.610000     1.760000     0.008000    29.450000   
max       0.200000    19.830000     2.930000     0.010000    60.010000   

            radium     selenium       silver      uranium  
count  7997.000000  7997.000000  7997.000000  7997.000000  
mean      2.919992     0.049679     0.147729     0.044677  
std       2.322833     0.028771     0.143525     0.026904  
min       0.000000     0.000000     0.000000     0.000000  
25%       0.820000     0.020000     0.040000     0.020000  
50%       2.410000     0.050000     0.080000     0.050000  
75%       4.670000     0.070000     0.240000     0.070000  
max       7.990000     0.100000     0.500000     0.090000

10. Проведен анализ с помощью функций describe в датасете, который вывел для каждого столбца свои статические показатели, к примеру, видно, что максимальное значение - в столбце "aluminium" = 5.050000, а а минимально значение в данном столбце = 0.000000.

In [61]:
subset = waterQuality1.loc[[0, 4, 3], ['aluminium', 'arsenic', 'barium']]
subset

aluminium  arsenic  barium
0       1.65     0.04    2.85
4       0.92     0.03    0.20
3       1.36     0.04    2.96

11. Проведена выборка данных по строкам и столбцам с помощью loc, то есть в начале были указаны индексы строк =, которые необходимо вывести, а затем столбцы, чьи значения можно тнпнрь сравнить из новой таблицы.

In [63]:
waterQuality1.to_csv("waterQuality1_clear.csv")

12. Сохранение нового датасета

Вывод: был реализован базовый анализ датасета: был выведен весь датасет и описан, выведена размерноть массива, выведены наименования всех колонок, написано что они означают в датасете, выведены все уникальные значения,отсортированы по определенным параметрам, удалены ненужные столбцы или строчки, заменены определенные значения в датасете, удалены дубликаты, проведен анализ с помощью функций info, проведен анализ с помощью функций describe, проведена выборку данных по строкам и столбцам с помощью loc, сохранен новый датасет, сделан вывод по каждому пункту и описать все.